### ISSUE TITLE
Fapi: profile_dir double free

### ISSUE THREAD

**Issue by joholl** [tpm2-software/tpm2-tss#2092]
This is a weird one. When I use the RSA profile, and call the encrypt/decrypt Fapi calls, I get a double free.

The first free seems to be in `iesys_cryptossl_hmac_finish()`: `SAFE_FREE(mycontext);`

The second free seems to be in `Fapi_Finalize()`: `SAFE_FREE((*context)->config.profile_dir);`

I hope you can reproduce this. To find these, I have `SAFE_FREE` redefined:

```c
#define SAFE_FREE(S) if((S) != NULL) {\
        LOG_ERROR("SAFE_FREE: " #S "=%p", (S));\
        free((void*) (S));\
        (S)=NULL;\
    }
```

Then I search for the freed memory address in the log. Commenting out either the first or the second free makes the error disappear.

**Comment by JuergenReppSIT** [comment]
@joholl The is an integration test which uses the encrypt/decrypt Fapi calls:
test/integration/fapi-data-crypt-rsa.fint (The source is test/integration/fapi-data-crypt.int.c)
 If you compare your use case with this test do you see any difference?

**Comment by joholl** [comment]
@JuergenReppSIT Sorry for the delay. I'm still not 100% sure what is happening, but I think I got a clue. I think I was wrong about the first free.

The first free seems to be in `Fapi_Encrypt.c`: `SAFE_FREE(command->out_data);`

The second free seems to be in `Fapi_Finalize()`: `SAFE_FREE((*context)->config.profile_dir);`

If I remove either of these frees, there will be no core dump.

I'm not sure why `command->out_data` exists in the first place. It does not seem to be used anywhere (or point to allocated memory for that matter).

**Comment by JuergenReppSIT** [comment]
@joholl Thank you for locating the problem. out_data was used in a former version were symmetric encryption was supported. I forgot to delete the field and free after the symmetric encryption was removed. I will create a PR.

**Comment by joholl** [comment]
Fixed by #2142
See also #2143, #2144, #2145

**Comment by joholl** [comment]
Resolved.

Thanks @JuergenReppSIT!
